src/auth/auth.service.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ForbiddenException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { UserService } from 'src/user/user.service';
+import { JwtService } from '@nestjs/jwt';
+import { User } from '@prisma/client';
+import { bcryptConstants } from './constants';
+import * as bcrypt from 'bcrypt';
+import { AuthDto } from './dto';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private userService: UserService,
+    private jwtService: JwtService,
+  ) {}
+
+  async signIn(user: User): Promise<{ access_token: string }> {
+    const payload = { email: user.email, password: user.password };
+    return {
+      access_token: await this.jwtService.signAsync(payload),
+    };
+  }
```

Comment:
If you follow my comment above ☝🏻 , then you can change it to:
```suggestion
 signIn(user: User): Promise<string> {
    const payload = { email: user.email, password: user.password };
    return this.jwtService.signAsync(payload);
  }
```

Note the lack of `await` next to `this.jwtService.signAsync(payload)` - that's because we can await this one level higher - in controller.  I also removed `async` as it's now not needed (no `await`'s). Note that the return type is still `Promise`, because even without `await` we will get `Promise` here, just a pending one (that will be resolved one level higher).